service/controller/v5/cloudconfig/cloud_config.go

```diff
@@ -63,6 +74,11 @@ func New(config Config) (*CloudConfig, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.AzureConfig.%s", config, err)
 	}
 
+	encrypter, err := NewEncrypter()
```

Comment:
Same as with the certs searcher. This dependency should be configured at runtime and not magically be used internally. Think about unit tests where you want to test different aspects of the template rendering. You may want to verify rendering works as expected without having to deal with encrypted content. Further the encryption protocol is now hard coded so there is no chance to get around it during testing. You may want to have a test encrypter which does not encrypt at all so you can verify the rendered templates. Then one aspect of unit testing further could be to also verify that the encryption with different specifications and requirements works as expected.

Reply:
I'll address this in separate PR